app/assets/javascripts/top_menu.js

```diff
@@ -241,8 +241,39 @@
       top_menus.push(new_menu);
     });
   };
+
+  // expands .search_btn inside the given wrapper
+  // to an input .search_field with the button appended to the right.
+  $.fn.expandableSearch = function () {
+    var wrapper = $(this)
+      , form = wrapper.closest('form')
+      , btn = wrapper.find('.search_btn')
+      , input = wrapper.find('.search_field')
+      , evtns = 'click.expandableSearch';
```

Comment:
I think it was `gulp lint` or jshint that can give you pointers.

And I shalleth not careth what them NPM heathens doeth!